scripts/ci.js

```diff
@@ -22,6 +25,74 @@ const execaOpts = {
   }
 };
 
+const isHeadCommitOfPR = async () => {
+  const prNumber = process.env.CODEBUILD_SOURCE_VERSION.replace('pr/', '');
+
+  const { stdout } = await execa('curl', [
+    '-H',
+    '"Authorization:',
+    'token',
+    `${process.env.GITHUB_TOKEN}"`,
+    `https://api.github.com/repos/FormidableLabs/badges/pulls/${prNumber}/commits`
+  ]);
+
+  const headCommit = JSON.parse(stdout).pop().sha;
+  return headCommit === process.env.CODEBUILD_RESOLVED_SOURCE_VERSION;
+};
+
+// eslint-disable-next-line promise/avoid-new
+const wait = ms => new Promise(resolve => setTimeout(resolve, ms));
+
+const slsDeployWithRetry = async (attempt = 0) => {
+  if (attempt > SLS_DEPLOY_RETRY_MAX_ATTEMPTS) {
+    throw new Error('Reached max retries for sls deploy. Giving up...');
+  }
+
+  if (attempt > 0) {
+    // eslint-disable-next-line no-console
+    console.log('Waiting to retry sls deploy...');
+    await wait(SLS_DEPLOY_RETRY_SLEEP_INTERVAL);
+  }
+
+  if (!(await isHeadCommitOfPR())) {
+    throw new Error('Commit is not at head of PR branch. Aborting...');
+  }
+
+  try {
+    const subprocess = execa(
+      'yarn',
+      ['sls', 'deploy', '--package', '.serverless'],
+      { ...execaOpts, stdio: 'pipe', all: true }
```

Comment:
Would stdio inherit her simpler and avoid the manual piping below?

Reply:
The issue with `inherit` is that `execa` doesn't then capture stdout/stdio and make them available to me for inspection. Here is a quick proof:

```
const execa = require('execa');

(async () => {

  try {
    await execa('terraform', ['foo', 'bar'], { stdio: 'inherit' });
  } catch (err) {
    console.log(err);
  }
})();
```

The console log output. Note that stdout and stderr are empty:
``` 
{ Error: Command failed with exit code 127: terraform foo bar
    at makeError (/Users/kevinmstephens/badges/node_modules/execa/lib/error.js:56:11)
    at handlePromise (/Users/kevinmstephens/badges/node_modules/execa/index.js:114:26)
    at <anonymous>
    at process._tickCallback (internal/process/next_tick.js:189:7)
  command: 'terraform foo bar',
  exitCode: 127,
  signal: undefined,
  signalDescription: undefined,
  stdout: undefined,
  stderr: undefined,
  failed: true,
  timedOut: false,
  isCanceled: false,
  killed: false }
```

Reply:
inherit _definitely_ works for stdout. Change to `await execa('echo', ['foo', 'bar'], { stdio: 'inherit' });` and you'll see stdout in the shell.

Not sure why stderr doesn't get caught. As this is a small thing, we can just leave as-is without researching more is my vote.

Reply:
Yeah you can see stdout and stderr in the shell but the error object execa returns doesn't give me the the stdout and stderr strings to then inspect programmatically. 

Tyler suggested using the aws sdk cloud formation waitFor method which looks like it will work. This whole execa call goes away then.